app/src/main/java/com/rockthevote/grommet/ui/views/AddressView.java

```diff
@@ -234,25 +275,45 @@ protected void dispatchRestoreInstanceState(SparseArray<Parcelable> container) {
         dispatchThawSelfOnly(container);
     }
 
+    /**
+     * should only trigger on county adapter update, which should only be possible if
+     * PA is the selected state
+     */
+    private void validateZipCode() {
+        String chosenCounty = countySpinner.getEditText().getText().toString();
+        boolean zipcodeInCounty = !chosenCounty.isEmpty() &&
+                counties.get(chosenCounty).contains(zipEditText.getText().toString());
+
+        zipTIL.setError(zipcodeInCounty ?
+                null : getContext().getString(R.string.zip_code_error));
+    }
+
     public Observable<Boolean> verify() {
 
-        unitTypeSpinner.setError(null);
-        unitTIL.setError(null);
 
-        String unit = unitEditText.getText().toString();
-        String type = unitTypeSpinner.getEditText().getText().toString();
+        if (zipTIL.getError() != null) {
+            // remember there are three of these that are evaluated at the same time in the PersonalInfoFragment
+            // these should probably not be validated if they are not used
+            return Observable.just(false);
+        } else {
+            unitTypeSpinner.setError(null);
+            unitTIL.setError(null);
 
-        boolean typeValidation = (Strings.isBlank(unit) || !Strings.isBlank(type));
-        if (!typeValidation) {
-            unitTypeSpinner.setError(getContext().getString(R.string.required_field));
-        }
+            String unit = unitEditText.getText().toString();
+            String type = unitTypeSpinner.getEditText().getText().toString();
 
-        boolean unitValidation = (!Strings.isBlank(unit) || Strings.isBlank(type));
-        if (!unitValidation) {
-            unitTIL.setError(getContext().getString(R.string.required_field));
-        }
+            boolean typeValidation = (Strings.isBlank(unit) || !Strings.isBlank(type));
```

Comment:
So the `unit` is supposed to be blank here?

Reply:
Yes, if the unit isn't blank then the type shouldn't be blank.

admittedly this is awful boolean logic.

Reply:
well, confusing boolean logic.

Either way, it's legacy and works, so i think we should leave it alone